beacon/handlers.go

```diff
@@ -71,6 +72,9 @@ func (pd *ProtocolDriver) handleProposal(ctx context.Context, peer p2p.Peer, msg
 		return errMalformedMessage
 	}
 
+	latency := receivedTime.Sub(pd.bClock.proposalSendTime(m.EpochID))
+	metrics.ReportMessageLatency(pubsub.BeaconProposalProtocol, pubsub.BeaconProposalProtocol, latency)
```

Comment:
i think it would be useful to have Beacon as a protocol, and Proposal/First/Followup/Weakcoin as a message type (the same as in hare). it will not be consistent with handlers (maybe we can change that too btw), but overall it will be easier to use

Reply:
Ok, yeah it did feel a bit odd to me to have 4 protocols for beacon, if we want to have all messages in the same protocol then we'll need to manage all messages with a single handler. I'll take a look and see what that change would entail.

Reply:
i think those can be separate changes though. we don't need to change handlers in order to have metrics

Reply:
Sure, it looks like there's a fair bit of work to merge the handlers so I'll add a ticket for that.